ext/hiera.yaml

```diff
@@ -2,14 +2,8 @@
 :backends:
   - yaml
 :hierarchy:
-  - defaults
-  - "%{clientcert}"
-  - "%{environment}"
-  - global
+  - "nodes/%{clientcert}"
+  - common
 
 :yaml:
-# datadir is empty here, so hiera uses its defaults:
-# - /etc/puppetlabs/code/hieradata on *nix
-# - %CommonAppData%\PuppetLabs\code\hieradata on Windows
-# When specifying a datadir, make sure the directory exists.
-  :datadir:
+  :datadir: "/etc/puppetlabs/code/environments/%{environment}/hieradata"
```

Comment:
And on 2003 `C:\Documents and Settings\All Users\Application Data\PuppetLabs\code`, which is why it's easier to implement this logic in code, and provide a `--configprint` like option to hiera to have it print what the defaults are.